### ISSUE TITLE
React v16 incompatible with react-addons-shallow-compare

### ISSUE THREAD

**Issue by zenyr** [Dean177/react-native-json-tree#135]
Current react-native-json-tree is very useful for debugging data structure. However it stopped working with newer projects with rn 0.43. It turns out that react 16 is not compatible with `react-addons-shallow-compare` module and forces to extend `React.PureComponent` instead. This is going to break many 3rd party react apps and unfortunately RNJT is not an exception. 

I'm filing this issue to make a discussion consolidating fixes/workarounds and such. 😄 

version: `react-native@0.43` (comes with `react@16 alpha 6`)

error:
```
error: bundling: UnableToResolveError: Unable to resolve module `react/lib/shallowCompare` from
 `~~\client\node_modules\react-addons-shallow-compare\index.js`: 
 Module does not exist in the module map or in these directories:
 ~~\client\node_modules\react\lib
```

**Comment by Dean177** [comment]
Thanks @zenyr  I will try to take a look at this soon

**Comment by Dean177** [comment]
I have released 1.0.0 to npm so I am closing this issue, please open a new one if you have any problems with the new version.
